### PR TITLE
8343137: C2: VerifyLoopOptimizations fails with "Was reachable in only one"

### PR DESCRIPTION
**Pull request by chhagedorn**
[JDK-8342043](https://bugs.openjdk.org/browse/JDK-8342043) introduced a new usage of `igvn.intcon()` but missed to call `set_ctrl()`. This simple patch fixes this.

While working on this, I've noticed that a few Assertion Predicates tests ended up in the `predicates` directory while they would have better been placed into `predicates/assertion`. It's probably not worth to file a separate issue just for that, so I've squeezed this trivial move into this fix. Note that `AssertionPredicateDoesntConstantFold` does not define a package and thus does not need an update to the command line.

Thanks,
Christian

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343137](https://bugs.openjdk.org/browse/JDK-8343137): C2: VerifyLoopOptimizations fails with "Was reachable in only one" (**Bug** - P3)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21739/head:pull/21739` \
`$ git checkout pull/21739`

Update a local copy of the PR: \
`$ git checkout pull/21739` \
`$ git pull https://git.openjdk.org/jdk.git pull/21739/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21739`

View PR using the GUI difftool: \
`$ git pr show -t 21739`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21739.diff">https://git.openjdk.org/jdk/pull/21739.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21739#issuecomment-2441497260)

### CODE REVIEW

File: src/hotspot/share/opto/loopnode.cpp [hunk 1]
```diff
@@ -4488,7 +4488,9 @@ void PhaseIdealLoop::eliminate_useless_template_assertion_predicates(Unique_Node
     OpaqueTemplateAssertionPredicateNode* opaque_node =
         C->template_assertion_predicate_opaq_node(i - 1)->as_OpaqueTemplateAssertionPredicate();
     if (!useful_predicates.member(opaque_node)) { // not in the useful list
-      _igvn.replace_node(opaque_node, _igvn.intcon(1));
+      ConINode* one = _igvn.intcon(1);
+      set_ctrl(one, C->root());
```
Comment (chhagedorn) [inline src/hotspot/share/opto/loopnode.cpp:4492]: Noticed that we find this pattern quite often in our code. Would be nice to have a `PhaseIdealLoop::intcon()` which calls `igvn.intcon()` and takes care of setting ctrl. I filed [JDK-8343148](https://bugs.openjdk.org/browse/JDK-8343148) to keep track of that.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

File: src/hotspot/share/opto/loopnode.cpp (hunk 1 above)
Comment (vnkozlov) [reply src/hotspot/share/opto/loopnode.cpp:4492]: Agree.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by chhagedorn** [comment]
Thanks Vladimir for your review!

**Pull request closed without merging**